scripts/prepdocslib/markdownparser.py

```diff
@@ -0,0 +1,30 @@
+import re
+from typing import IO, AsyncGenerator
+
+from .page import Page
+from .parser import Parser
+
+
+def cleanup_data(data: str) -> str:
+    """Cleans up the given content using regexes
+    Args:
+        data: (str): The data to clean up.
+    Returns:
+        str: The cleaned up data.
+    """
+    # match two or more newlines and replace them with one new line
+    output = re.sub(r"\n{2,}", "\n", data)
+    # match two or more spaces that are not newlines and replace them with one space
+    output = re.sub(r"[^\S\n]{2,}", " ", output)
+
+    return output.strip()
+
+
+class MarkdownParser(Parser):
```

Comment:
Does this currently do anything Markdown specific? I'm wondering if it should just be called a TextParser, and also be used for .txt formats. Did you check if azure-search-openai-javascript does any Markdown-specific processing?

Reply:
Thanks, naming it TextParser does make more sense, it's also similar to what `azure-search-openai-javascript` [have done](https://github.com/Azure-Samples/azure-search-openai-javascript/blob/main/packages/indexer/src/lib/formats/text.ts).

Reply:
I have also added a mapping for `.txt` files to use that parser as well